src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -58,51 +65,44 @@ fun buildPreparableFiltersSql(
         metadataOptions: Set<MetadataOption>,
         linking: Boolean,
         idsPresent: Boolean,
-        partitionsPresent: Boolean
+        partitionsPresent: Boolean,
+        detailed: Boolean = false
 ): Pair<String, Set<SqlBinder>> {
     val filtersClauses = buildPreparableFiltersClause(startIndex, propertyTypes, propertyTypeFilters)
     val filtersClause = if (filtersClauses.first.isNotEmpty()) " AND ${filtersClauses.first} " else ""
-
     val metadataOptionColumns = metadataOptions.associateWith(::mapMetaDataToColumnSql)
-    val nonAggregatedMetadataSql = metadataOptionColumns
-            .filter { !isMetaDataAggregated(it.key) }.values.joinToString("")
-    val innerGroupBy = groupBy(ESID_EKID_PART_PTID + nonAggregatedMetadataSql)
-    // TODO: remove IS NOT NULL post-migration
-    val linkingClause = if (linking) {
-        " AND ${ORIGIN_ID.name} IS NOT NULL AND ${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID.id}' "
-    } else {
-        ""
-    }
+    val metadataOptionColumnsSql = metadataOptionColumns.values.joinToString("")
+    val innerGroupBy = groupBy(ESID_EKID_PART_PTID)
+    val outerGroupBy = if (metadataOptions.contains(MetadataOption.ORIGIN_IDS)) {
+        groupBy("$ESID_EKID_PART,${ORIGIN_ID.name}")
+    } else groupBy(ESID_EKID_PART)
+    val linkingClause = if (linking) " AND ${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID}' " else ""
 
     val innerSql = selectEntitiesGroupedByIdAndPropertyTypeId(
             metadataOptions,
             idsPresent = idsPresent,
-            partitionsPresent = partitionsPresent
+            partitionsPresent = partitionsPresent,
+            detailed = detailed
     ) + linkingClause + filtersClause + innerGroupBy
 
-    val metadataOptionColumnsSql = metadataOptionColumns
-            .values.joinToString("")
-    // TODO don't filter out LAST_WRITE from group by after empty rows are eliminated  https://jira.openlattice.com/browse/LATTICE-2254
-    val metadataOptionOuterGroupBySql = metadataOptionColumns
-            .filter { it.key != MetadataOption.LAST_WRITE }
-            .values.joinToString("")
-    val outerGroupBy = groupBy(ESID_EKID_PART + metadataOptionOuterGroupBySql)
-    val sql = "SELECT ${ENTITY_SET_ID.name},${ID_VALUE.name},${PARTITION.name}$metadataOptionColumnsSql,$jsonValueColumnsSql " +
+    val sql = "SELECT ${ENTITY_SET_ID.name},${ID_VALUE.name},${PARTITION.name}$metadataOptionColumnsSql," +
+            "jsonb_object_agg($PROPERTY_TYPE_ID,PROPERTIES) " +
```

Comment:
`PROPERTY_TYPE_ID.name` should be here I think

Reply:
+1, and also `$PROPERTIES`